### PR TITLE
[MIG] product_supplier_code_purchase: Migration to 14.0

### PR DESCRIPTION
**Pull request by Zar21**
Standard migration Odoo 14.0

### CODE REVIEW

**Comment by HviorForgeFlow** [comment]
Commit history is not properly set up, please follow [this guide,](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0) replacing 13.0 for 12.0 allow you to preserve full commit history.
Take into account that you need to fetch branch 12 before do the `git format-patch`

**Comment by Zar21** [comment]
Commit history repaired

**Review by JuanyDForgeflow — APPROVED** [review]
LGTM

File: product_supplier_code_purchase/models/purchase_order.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    def _onchange_product_code(self):
+        for line in self:
+            supplier_info = line.product_id.seller_ids.filtered(
+                lambda s: (
+                    s.product_id == line.product_id and s.name == line.partner_id
+                )
+            )
+            if supplier_info:
+                code = supplier_info[0].product_code or ""
+                line.product_supplier_code = code
```
Comment (MeetKD) [inline product_supplier_code_purchase/models/purchase_order.py]:
```suggestion
                line.product_supplier_code = supplier_info[0].product_code or ""
```

File: product_supplier_code_purchase/models/purchase_order.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            else:
+                supplier_info = line.product_id.seller_ids.filtered(
+                    lambda s: (
+                        s.product_tmpl_id == line.product_id.product_tmpl_id
+                        and s.name == line.partner_id
+                    )
+                )
+                if supplier_info:
+                    code = supplier_info[0].product_code or ""
+                    line.product_supplier_code = code
```
Comment (MeetKD) [inline product_supplier_code_purchase/models/purchase_order.py]:
```suggestion
                line.product_supplier_code = supplier_info[0].product_code or ""
```

**Review by MeetKD — CHANGES_REQUESTED** [review]
Minor Suggestion

**Review by AaronHForgeFlow — APPROVED** [review]
Functional review LGTM

Can you please attend comments?

**Comment by ferran-S73** [comment]
> Functional review LGTM
> 
> Can you please attend comments?

Hi. we're going to check the suggestions and apply any necessary changes as soon as possible. Will keep you updated

**Comment by ferran-S73** [comment]
@MeetKD @AaronHForgeFlow I've just pushed an update with the last changes proposed. Hope everything is alright now!

**Pull request merged**